src/dist/replication/lib/replica_stub.cpp

```diff
@@ -902,6 +902,61 @@ void replica_stub::on_query_replica_info(const query_replica_info_request &req,
     resp.err = ERR_OK;
 }
 
+// ThreadPool: THREAD_POOL_DEFAULT
+void replica_stub::on_query_disk_info(const query_disk_info_request &req,
+                                      /*out*/ query_disk_info_response &resp)
+{
+    resp.err = ERR_OK;
+
+    for (const auto &dir_node : _fs_manager._dir_nodes) {
+        disk_info info;
+        // app_name empty means query all app replica_count
+        if (req.app_name.empty()) {
+            for (const auto &holding_primary_replicas : dir_node->holding_primary_replicas) {
+                info.holding_primary_replica_counts[holding_primary_replicas.first] =
+                    static_cast<int>(holding_primary_replicas.second.size());
+            }
+
+            for (const auto &holding_secondary_replicas : dir_node->holding_secondary_replicas) {
+                info.holding_secondary_replica_counts[holding_secondary_replicas.first] =
+                    static_cast<int>(holding_secondary_replicas.second.size());
+            }
+        } else {
+            int app_id = 0;
+            {
+                zauto_read_lock l(_replicas_lock);
+                if (!(app_id = get_app_id_from_replicas(req.app_name)) &&
```

Comment:
You will acquire the lock several times, why not move this judgement outside of the loop?